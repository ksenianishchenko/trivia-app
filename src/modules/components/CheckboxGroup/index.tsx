import React from "react";
import { TriviaQuestionItemAnswersChoice } from "../../../abstractions/api/models/triviaQuestionItem";

type CheckboxGroupTypes = {
  option: TriviaQuestionItemAnswersChoice;
  index: number;
}

const CheckboxGroup = (props: CheckboxGroupTypes) => {
  const {option, index} = props;

  return <div className="checkbox-group" key={index}>
    <label className="checkbox-group__label" htmlFor={option.id}>
        <input
            className="checkbox-group__input"
            type="checkbox"
            name={option.text}
            id={option.id}
        />
        <span>{option.text}</span>
    </label>
  </div>
}

export default CheckboxGroup;