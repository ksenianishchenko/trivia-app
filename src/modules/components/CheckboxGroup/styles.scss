.checkbox-group.error {
  color: crimson;
}

.checkbox-group.correct {
  color: darkgreen;
}