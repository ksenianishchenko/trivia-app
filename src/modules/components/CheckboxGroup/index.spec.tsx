import React from "react";
import { shallow } from 'enzyme';
import CheckboxGroup from "./index";


describe("should render <CheckboxGroup /> component correct", () => {

  const mockOption = {
    "id": "0",
    "text": "Lord Volan De Mort",
    "imageUrl": ""
  }

  it("should render checkbox group button", () => {
      const tree = shallow(<CheckboxGroup option={mockOption} index={1} />);
      expect(tree).toMatchSnapshot();
  })

})