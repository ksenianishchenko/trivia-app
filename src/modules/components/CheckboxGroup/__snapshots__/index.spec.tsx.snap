// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <CheckboxGroup /> component correct should render checkbox group button 1`] = `
<div
  className="checkbox-group"
  key="1"
>
  <label
    className="checkbox-group__label"
    htmlFor="0"
  >
    <input
      className="checkbox-group__input"
      id="0"
      name="Lord Volan De Mort"
      type="checkbox"
    />
    <span>
      Lord Volan De Mort
    </span>
  </label>
</div>
`;
